packages/legend-query/src/stores/filterOperators/QueryBuilderFilterOperator_Equal.ts

```diff
@@ -144,22 +164,33 @@ export class QueryBuilderFilterOperator_Equal extends QueryBuilderFilterOperator
             return enumValueInstanceValue;
           }
           throw new UnsupportedOperationError(
-            `Can't get default value for filter operator '${this.getLabel(
-              filterConditionState,
-            )}' since enumeration '${propertyType.path}' has no value`,
+            `Can't get default value for filter operator '${this.getLabel()}' since enumeration '${
+              propertyType.path
+            }' has no value`,
           );
         }
         throw new UnsupportedOperationError(
-          `Can't get default value for filter operator '${this.getLabel(
-            filterConditionState,
-          )}' when the LHS property is of type '${propertyType.path}'`,
+          `Can't get default value for filter operator '${this.getLabel()}' when the LHS property is of type '${
+            propertyType.path
+          }'`,
         );
     }
   }
 
   buildFilterConditionExpression(
     filterConditionState: FilterConditionState,
   ): ValueSpecification {
+    if (
+      filterConditionState.propertyExpressionState.propertyExpression.func
```

Comment:
this is only applied to `DateTime` not date?

Reply:
Since Date is the supertype, RHS could be date/datetime/strictdate,  etc.
For DateTime, if RHS is date/stricdate, isOnDay() needs to be called.